frontend/src/metabase/selectors/settings.ts

```diff
@@ -1,28 +1,31 @@
 import { createSelector } from "@reduxjs/toolkit";
 
-import type {
-  SettingKey,
-  Settings,
-  TokenStatus,
-  Version,
-} from "metabase-types/api";
+import type { TokenStatus, Version } from "metabase-types/api";
 import type { State } from "metabase-types/store";
 import { getPlan } from "metabase/common/utils/plan";
 
-export const getSettings = createSelector(
-  (state: State) => state.settings,
-  settings => settings.values,
-);
+export const getSettings: <S extends State>(state: S) => GetSettings<S> =
+  createSelector(
+    (state: State) => state.settings,
+    settings => settings.values,
+  );
 
 export const getSettingsLoading = createSelector(
   (state: State) => state.settings,
   settings => settings.loading,
 );
 
-export const getSetting = <T extends SettingKey>(
-  state: State,
+type GetSettings<S extends State> = S["settings"]["values"];
+type GetSettingKey<S extends State> = keyof GetSettings<S>;
+
+export const getSetting = <S extends State, T extends GetSettingKey<S>>(
```

Comment:
Makes this selector generic so that it will allow to use the enterprise setting keys which don't exist in OSS state.

It [was used here.](https://github.com/metabase/metabase/pull/38850/commits/1587411da685857ee586b3096a58148f6a0ce36c#diff-ce1e941a646242ac9341073cc9777328968aced31361ff7fc699502ab8352fabR35) But that change is already reverted. So I only keep the type improvement.